### PR TITLE
Set APPEND_PROJECT_NAME_TO_INCLUDEDIR to ON

### PR DESCRIPTION
**Pull request by sloretz**
Part of ros2/ros2#1150 and follow up from https://github.com/eProsima/Fast-DDS/pull/2535

### CODE REVIEW

**Comment by clalancette** [comment]
I have what is maybe a dumb question.  Should we be patching this branch directly, or should we maybe be adding a new patch to the https://github.com/ros2-gbp/fastrtps-release/tree/patches/debian/rolling/fastrtps branch?  I see "SECURITY=ON" in both branches, so it is not clear to me which way to go.  Maybe @cottsay or @nuclearsandwich can clarify here?

**Comment by cottsay** [comment]
> Should we be patching this branch directly, or should we maybe be adding a new patch to the `patches/debian/rolling/fastrtps` branch?

tl;dr - never change the `patches/*` branches directly.

Bloom manages that branch with `git-bloom-patch`. Always apply the changes to the branch you want patched.

Example documentation: http://wiki.ros.org/bloom/Tutorials/ReleaseThirdParty#Adding_an_Install_Rule_as_a_Patch

**Comment by clalancette** [comment]
> tl;dr - never change the `patches/*` branches directly.

Thanks!  That is exactly the information I was looking for.

**Review by clalancette — APPROVED** [review]
This looks good to me, thanks.
